### PR TITLE
Ramya teams limit see all 

### PR DESCRIPTION
**Pull request by scoopstroop92**
# Description
- A “Limit See-All” option for teams 

-  Other Links ->Teams

- When this option is turned on for a team member,  the team member  shouldn't  be able to see  other members of the team in their dashboard 

- The member however should be able to see team members of other teams  they are part of 

## Related PRS (if any):
This backend PR is related to the [1425](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425) frontend PR.
To test this backend PR you need to checkout the [1425](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425) frontend PR.

## Main changes explained:

- Added  a "visible" attribute to Member Object in Teams document and set it to true by default.
- Added a new route to updating the team member visibility. 
- Add  a  function  called  "updateTeamVisibility "in teams Controller   to process the newly selected visiblity of the team member

## How to test:
1. check into current branch
2. run node src/utilities/addMembersToTeams.js so that "visible" attribute gets added to all existing team members
3. run `npm install ` followed by `npm run build start`  and  `npm start ` to run this PR locally
4.  Now follow the steps mentioned in the front end PR

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/89441435/887ed506-2ce6-443d-beb4-645abc471335

## Note:
- Kindly mention any  bugs , errors you see in the console. 
- Try breaking the functionality , any suggestion is helpful

### CODE REVIEW

**Review by DavidC0126 — APPROVED** [review]
LGTM, commented on FE PR.

**Review by shubhankarval — CHANGES_REQUESTED** [review]
Left a comment on FE [1425](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425#pullrequestreview-1688160585).

**Review by StrawberryCalpico — CHANGES_REQUESTED** [review]
Hi! I reviewed your PR, left comment on the [FR#1425](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425#pullrequestreview-1691026115)

**Review by ChirayuRai — COMMENTED** [review]
Hello! I added my comments in FE#1425!

**Comment by ramyaram2092** [comment]
I am investigating the bug/ feedbacks received so far. Marking the PR to "do not review " for now on the same account

**Review by Alforoan — APPROVED** [review]
I left a comment on the FE pr.

**Review by ljrirene — COMMENTED** [review]
Please see my review of #1425 here: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425#pullrequestreview-1795511374

Thank you!

**Comment by yuhuimin1996** [comment]
Please see my comments at the FE pr #1425

File: src/controllers/teamController.js [hunk 1]
```diff
@@ -1,22 +1,27 @@
 const mongoose = require('mongoose');
 const userProfile = require('../models/userProfile');
+// const teamDetails=require('../models/team');
```
Comment (Renanluizssx) [inline src/controllers/teamController.js:3]: you could delete this comment

**Review by Renanluizssx — COMMENTED** [review]
hi @ramyaram2092, please see my review at the PR #1425: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1425

**Comment by ramyaram2092** [comment]
Hi all thanks for your detailed feedback . I am looking into your feedback and addressing them, marking the PR do not review on the same note

**Comment by ramyaram2092** [comment]
Addressed the feedbacks at [731](https://github.com/OneCommunityGlobal/HGNRest/pull/731)

**Pull request closed without merging**